CONTRIBUTING.md

```diff
@@ -0,0 +1,466 @@
+   General best practice guidelines on how to develop with Substrate.
+   If you create best practice documentation about using a specific
+   feature of Substrate, you should create the documentation in the same
+   directory as the other documentation for that specific feature.
+
+* Substrate Rust Documentation - `substrate.dev/rustdocs/`
+
+   Auto-generated Rust documentation from the Substrate repository. See [Documentation Comments](#documentation-comments).
+
+* Substrate Developer Documentation - `substrate.dev/docs/`
+
+   In the `/docs/` directory, you should create your
+   documentation or images in one of these sub-directories:
+    
+   *  `development`
+        Instructions, tutorials, and procedural documentation for developers
+        who are working on Substrate. This directory includes documentation
+        on how to get started, build, run, and test Substrate. You should organize the content
+        that you create by specific activities, such as testing, getting
+        started, or by workflow topic.
+    
+   * `conceptual`
+        Concept and developer guides about the features of Substrate. You
+        should organize the content that you create by specific features.
+    
+   * `images`
+        Images that are used in the documentation. You should place images in
+        this common directory and avoid placing images in the same directory
+        as documentation.
+
+### What documentation should I create?
+
+Most documentation can be divided into four categories:
+
+- [Reference](#reference-documentation) - Information-oriented documentation
+
+- [Conceptual](#conceptual-documentation) - Understanding-oriented
+  documentation
+
+- [Procedural](#procedural-documentation)
+
+    - How to - Goal-oriented documentation
+
+    - Tutorial - Learning-oriented documentation
+
+See [Documentation Types](#documentation-types) for more information.
+
+However, comments in your code are very important for maintainability and helping other people
+understand your code. See the [Code Comment Guidelines](#documentation-comments) for style guidelines
+related to comments for your code.
+
+### What documentation style guidelines should I follow?
+
+It is important to follow documentation style guidelines to ensure that the documentation
+created by a large number of contributors can flow together. See
+[Documentation Style Guide](#documentation-style).
+
+### How can I link to other docs in my documentation?
+
+Use absolute paths starting with `/` and ending with `.md`, like [`/getting-started/using-the-substrate-scripts.md`](/substrate/node-template/runtime/src/template.rs).
+Never use relative paths with ".." that point to content outside of `/docs`.
+
+### How can I link to source code in my documentation?
+
+Use the `/rustdocs` to link to specific parts of the Substrate source code, for example a module's `trait`. When referencing specific lines of code, copy those lines into the documentation. However, use this sparingly.
```

Comment:
I think it's worth establishing here that code shouldn't be referenced by line number. The exception might be comments to discuss docs, but maybe no docs?